warehouse/backend_config.go

```diff
@@ -222,6 +215,49 @@ func (s *backendConfigManager) processData(ctx context.Context, data map[string]
 	}
 }
 
+// namespace gives the namespace for the warehouse in the following order
+//  1. user set name from destinationConfig
+//  2. from existing record in wh_schemas with same source + dest combo
+//  3. convert source name
+func (s *backendConfigManager) namespace(ctx context.Context, source backendconfig.SourceT, destination backendconfig.DestinationT) string {
+	destType := destination.DestinationDefinition.Name
+	destConfig := destination.Config
+
+	if destType == whutils.CLICKHOUSE {
+		if database, ok := destConfig["database"].(string); ok {
+			return database
+		}
+		return "rudder"
+	}
+
+	if destConfig["namespace"] != nil {
+		namespace, _ := destConfig["namespace"].(string)
+		if len(strings.TrimSpace(namespace)) > 0 {
+			return whutils.ToProviderCase(destType, whutils.ToSafeNamespace(destType, namespace))
+		}
+	}
+
+	namespacePrefix := s.conf.GetString(fmt.Sprintf("Warehouse.%s.customDatasetPrefix", whutils.WHDestNameMap[destType]), "")
+	if namespacePrefix != "" {
+		return whutils.ToProviderCase(destType, whutils.ToSafeNamespace(destType, fmt.Sprintf(`%s_%s`, namespacePrefix, source.Name)))
+	}
+
+	namespace, err := s.schema.GetNamespace(ctx, source.ID, destination.ID)
+	if err != nil {
+		s.logger.Errorw("getting namespace",
```

Comment:
@achettyiitr we don't have enough context here, right? We should at least log the error.

Reply:
Good point. Let me make the changes in a separate PR.